tests/php/tests/includes/helper/test_class.wp-job-manager-com-auth-token.php

```diff
@@ -0,0 +1,263 @@
+		// Arrange.
+		$instance  = WP_Job_Manager_Com_Auth_Token::instance();
+		$instance2 = WP_Job_Manager_Com_Auth_Token::instance();
+
+		// Act.
+
+		// Assert.
+		$this->assertSame( $instance2, $instance );
+	}
+
+	public function testGenerate_WhenPassedInvalidObjectType_ShouldReturnError() {
+		// Arrange.
+		$instance = WP_Job_Manager_Com_Auth_Token::instance();
+
+		// Act.
+		$result = $instance->generate( 'comment', 1);
+
+		// Assert.
+		$this->assertTrue( is_wp_error( $result ) );
+		$this->assertEquals( 'wpjobmanager-com-invalid-type' , $result->get_error_code() );
+	}
+
+
+	public function testGenerate_WhenPassedInvalidObjectID_ShouldReturnError() {
+		// Arrange.
+		$instance = WP_Job_Manager_Com_Auth_Token::instance();
+
+		// Act.
+		$result = $instance->generate( 'user', "test");
+
+		// Assert.
+		$this->assertTrue( is_wp_error( $result ) );
+		$this->assertEquals( 'wpjobmanager-com-token-not-saved' , $result->get_error_code() );
+	}
+
+	public function testGenerate_WhenPassedUser_ShouldPersistMeta() {
+		// Arrange.
+		$instance = WP_Job_Manager_Com_Auth_Token::instance();
+		$user = $this->factory->user->create_and_get();
+
+		// Act.
+		$result = $instance->generate( 'user', $user->ID);
+
+		// Assert.
+		$this->assertIsString( $result );
+		$this->assertNotEmpty( get_user_meta( $user->ID, WP_Job_Manager_Com_Auth_Token::META_KEY ) );
+	}
+
+	public function testGenerate_WhenPassedPost_ShouldPersistMeta() {
+		// Arrange.
+		$instance = WP_Job_Manager_Com_Auth_Token::instance();
+		$post = $this->factory->post->create_and_get();
+
+		// Act.
+		$result = $instance->generate( 'post', $post->ID);
+
+		// Assert.
+		$this->assertIsString( $result );
+		$this->assertNotEmpty( get_post_meta( $post->ID, WP_Job_Manager_Com_Auth_Token::META_KEY ) );
+	}
+
+	public function testValidate_WhenPassedInvalidObjectType_ShouldReturnFalse() {
+		// Arrange.
+		$instance = WP_Job_Manager_Com_Auth_Token::instance();
+
+		// Act.
+		$result = $instance->validate( 'comment', 1, 'test' );
+
+		// Assert.
+		$this->assertFalse( $result );
+	}
+
+	public function testValidate_WhenPassedInvalidObjectID_ShouldReturnFalse() {
+		// Arrange.
+		$instance = WP_Job_Manager_Com_Auth_Token::instance();
+
+		// Act.
+		$result = $instance->validate( 'user', 'test', 'test' );
+
+		// Assert.
+		$this->assertFalse( $result );
+	}
+
+	public function testValidate_WhenPassedUserWithoutMeta_ShouldReturnFalse() {
+		// Arrange.
+		$instance = WP_Job_Manager_Com_Auth_Token::instance();
+		$user = $this->factory->user->create_and_get();
+
+		// Act.
+		$result = $instance->validate( 'user', $user->ID, 'test' );
+
+		// Assert.
+		$this->assertFalse( $result );
+	}
+
+	public function testValidate_WhenPassedUserWithMetaButInvalidToken_ShouldReturnFalse() {
+		// Arrange.
+		$instance = WP_Job_Manager_Com_Auth_Token::instance();
+		$user = $this->factory->user->create_and_get();
+		$token = $instance->generate('user', $user->ID);
```

Comment:
```suggestion
		$token = $instance->generate('user', $user->ID );
```

Reply:
Fixed in 3e75085c236778ceb0fe8e4f055aca2b6a1323e3. Thanks!